src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -332,20 +365,27 @@ private void EnsureFileVersionProvider()
                    .Append("</script>");
         }
 
-        private bool ShouldAddFileVersion()
-        {
-            return FileVersion ?? false;
-        }
-
-        private void AppendAttribute(TagHelperContent content, string key, string value, bool escapteQuotes)
+        private void AppendAttribute(TagHelperContent content, string key, object value, bool escapeQuotes)
         {
             content
                 .Append(" ")
-                .Append(key)
-                .Append(escapteQuotes ? "=\\\"" : "=\"")
-                .Append(value)
-                .Append(escapteQuotes ? "\\\"" : "\"");
+                .Append(key);
+            if (escapeQuotes)
```

Comment:
Maybe a better name than `escapeQuotes`? Just because it also determines if we encode the `value`.